java/core/src/java/org/apache/orc/impl/DictionaryUtils.java

```diff
@@ -46,6 +47,18 @@ public static void getTextInternal(Text result, int position,
     byteArray.setText(result, offset, length);
   }
 
+  public static ByteBuffer getTextInternal(int position, DynamicIntArray keyOffsets,
+      DynamicByteArray byteArray) {
+    final int offset = keyOffsets.get(position);
+    final int length;
+    if (position + 1 == keyOffsets.size()) {
+      length = byteArray.size() - offset;
+    } else {
+      length = keyOffsets.get(position + 1) - offset;
+    }
```

Comment:
Length calculation is repeated at least 5 times already -- lets move this to a helper function

Reply:
I agree with @pgaref 's perspective. Could you elaborate a little bit more about your refactoring idea? Is that better?
>  I actually think this DictionaryUtils class can probably be removed and roll the functionality into StringHashTable directly.